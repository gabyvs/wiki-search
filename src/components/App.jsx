import React, { Component } from 'react';
import styled from 'styled-components';

const Header = styled.header`
  position: fixed;
  top: 0;
  width: 100%;
  height: 100px;
  border-bottom: 1px solid rgba(0,0,0,0.15);
  background-color: rgba(236, 236, 236, 0.3);
  display: flex;
  align-items: center;
  padding-left: 50px;
`;

const Main = styled.main`
  margin-top: 100px;
  overflow: auto;
`;

class App extends Component {
  render() {
    return (
      <div className="App">
        <Header />
        <Main />
      </div>
    );
  }
}

export default App;
