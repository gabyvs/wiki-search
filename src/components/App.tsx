import React     from 'react';
import SearchBar from './SearchBar';
import Result    from './Result';

const styles = {
  header: {
    top: 0,
    width: '100%',
    height: '10rem',
    borderBottom: '1px solid rgba(0,0,0,0.15)',
    backgroundColor: 'rgba(236, 236, 236, 0.3)',
    display: 'flex',
    alignItems: 'center',
    paddingLeft: '5rem',
    position: 'fixed' as 'fixed'
  },
  main: {
    marginTop: "10rem",
    overflow: "auto",
  }
};

const App = () => (
  <>
    <header style={styles.header}>
      <SearchBar />
    </header>
    <main style={styles.main}>
      <Result
        url="www.rei.com"
        title="Go to the wilderness"
        snippet="This summer experience the backcountry in the company of your beloved friends."/>
    </main>
  </>
);

export default App;
