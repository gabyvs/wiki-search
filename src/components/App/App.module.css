@import '../../theme.css';

.header {
  top: 0;
  width: 100%;
  height: 10rem;
  border-bottom: 1px solid var(--mat-divider);
  background-color: var(--dark-background);
  display: flex;
  align-items: center;
  padding-left: 5rem;
  position: fixed;
}

.main {
  margin: 10rem 0 0 12rem;
  width: 64rem;
  overflow: auto;
}

@media screen and (max-width: 800px) {
  .header {
    padding: 2rem 0 2rem;
    height: auto;
    justify-content: space-between;
    flex-direction: column;
  }

  .main {
    margin: 14.5rem auto 0;
    width: 80%;
  }
}