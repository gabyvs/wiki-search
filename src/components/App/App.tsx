import React       from 'react';
import SearchBar   from '../SearchBar/SearchBar';
import styles      from './App.module.css';
// import List        from '../List/List';
// import data        from '../../sampleResponse';
import { IResult } from '../../domain/result';
import Result      from '../Result/Result';
import List       from '../List/List';

const result: IResult = {
  title: "abc",
  snippet: "abc"
};
const results: IResult[] = [result];
const App = () => (
  <>
    <header className={styles.header}>
      <SearchBar/>
    </header>
    <main className={styles.main}>
      <Result {...result} />
      <Result snippet={"snippet"} title={"title"}/>
      <List results={results}/>
    </main>
  </>
);

export default App;
