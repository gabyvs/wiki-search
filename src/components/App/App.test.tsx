import React        from "react";
import {
  shallow,
  ShallowWrapper }  from "enzyme";
import App          from "./App";

describe("App", () => {
  let app: ShallowWrapper;

  beforeEach(() => {
    app = shallow(<App />);
  });

  it('renders without crashing', () => {});

  it("renders correctly", () => {
    expect(app).toMatchSnapshot();
  });
});
