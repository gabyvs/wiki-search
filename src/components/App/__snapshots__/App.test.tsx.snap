// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
<Fragment>
  <header
    className="header"
  >
    <SearchBar />
  </header>
  <main
    className="main"
  />
</Fragment>
`;
